sql/core/src/main/scala/org/apache/spark/sql/util/QueryExecutionListener.scala

```diff
@@ -76,7 +76,11 @@ trait QueryExecutionListener {
 class ExecutionListenerManager private[sql](session: SparkSession, loadExtensions: Boolean)
   extends Logging {
 
-  private val listenerBus = new ExecutionListenerBus(session)
+  private val listenerBus = new ExecutionListenerBus(session.sessionUUID)
+  session.sparkContext.listenerBus.addToSharedQueue(listenerBus)
+  session.sparkContext.cleaner.foreach { cleaner =>
+    cleaner.registerSparkListenerForCleanup(this, listenerBus)
+  }
```

Comment:
Manage this within `ExecutionListenerBus` instead of pulling it out ?
Make `sessionUUID` an instance variable - and use `session` only in constructor (so it is not a field in the class)

Reply:
Oh, this is a backporting PR. @Ngone51 can we address the comment in a separate PR so that followup can also be ported back?

Reply:
Sure, I'll do a follow-up. Thanks @mridulm

Reply:
Sorry, missed the original PR !